### PR TITLE
feature: support inputs in the device interface

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Support specifying inputs from `device.run` for the local simulator. A circuit with free parameters, when passed along with correct input, will run successfully on the local simulator. The AwsDevice class does not at this moment have an option for inputs added, so the behavior is unchanged.

*Testing done:*

Unit tests and manual e2e verification.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/devices/device.py [hunk 1]
```diff
@@ -33,7 +33,12 @@ def __init__(self, name: str, status: str):
 
     @abstractmethod
     def run(
-        self, task_specification: Union[Circuit, Problem], shots: Optional[int], *args, **kwargs
+        self,
+        task_specification: Union[Circuit, Problem],
+        shots: Optional[int],
+        inputs: Optional[Dict[str, float]],
```
Comment (speller26) [inline src/braket/devices/device.py:39]: Do we want to default here too?
```suggestion
        inputs: Optional[Dict[str, float]] = None,
```

**Review by speller26 — APPROVED** [review]

File: src/braket/devices/device.py (hunk 1 above)
Comment (ajberdy) [reply src/braket/devices/device.py:39]: Ya then we should probably do the same for shots

File: src/braket/devices/device.py [hunk 2]
```diff
@@ -33,7 +33,12 @@ def __init__(self, name: str, status: str):
 
     @abstractmethod
     def run(
-        self, task_specification: Union[Circuit, Problem], shots: Optional[int], *args, **kwargs
+        self,
+        task_specification: Union[Circuit, Problem],
+        shots: Optional[int] = 0,
```
Comment (speller26) [inline src/braket/devices/device.py:38]: This I'm not as convinced by, since QPUs are also based on this. No strong opinions though.

**Review by speller26 — APPROVED** [review]

File: src/braket/devices/device.py (hunk 2 above)
Comment (ajberdy) [reply src/braket/devices/device.py:38]: Then with that logic, task specification could be e.g. an annealing problem which won't take inputs, so maybe we shouldn't have defaults for either?

**Review by speller26 — APPROVED** [review]

**Pull request merged**